File: test/test_dataset_series.py
import yt
import sys
import os
import glob

test_dir = sys.argv[1]
file_to_write = os.path.join(test_dir, "dataset_series.h5")

fns = glob.glob("WindTunnel/windtunnel_4lev_hdf5_plt_cnt_[0-9][0-9]0*")
fns.sort()

time = []
max_dens = []
ts = yt.DatasetSeries(fns)

for ds in ts:
    sp = ds.sphere("c", (0.2,"unitary"))
    time.append(ds.current_time)
    max_dens.append(sp["density"].max())

time = yt.YTArray(time)
max_dens = yt.YTArray(max_dens)

time.write_hdf5(file_to_write, dataset_name="time")
max_dens.write_hdf5(file_to_write, dataset_name="max_dens")

File: test/test_dataset_series.jl
using YT
using Glob
using Base.Test

test_dir = Pkg.dir("YT") * "/test"

file_to_read = "$(test_dir)/dataset_series.h5"

fns = sort(glob("WindTunnel/windtunnel_4lev_hdf5_plt_cnt_[0-9][0-9]0*"))
time = YTQuantity[]
max_dens = YTQuantity[]
ts = DatasetSeries(fns)
for ds in ts
    append!(time, [ds.current_time])
    sp = Sphere(ds, "c", (0.2,"unitary"))
    append!(max_dens, [maximum(sp["density"])])
end
time = YTArray(time)
max_dens = YTArray(max_dens)

a = time
b = from_hdf5(file_to_read, dataset_name="time")
@test all(a.value .== b.value)
@test a.units == b.units

@test ts[1].current_time == a[1]

a = max_dens
b = from_hdf5(file_to_read, dataset_name="max_dens")
@test all(a.value .== b.value)
@test a.units == b.units

@test length(fns) == length(ts)
